docs/data/sdks/typescript-react-native/migration.md

```diff
@@ -3,6 +3,326 @@ title: React Native SDK Migration Guide
 description: Use this guide to easily migrate from Amplitude's maintenance React Native SDK (@amplitude/react-native) to the new SDK (@amplitude/analytics-react-native).
 ---
 
+Amplitude New React Native SDK (`@amplitude/analytics-react-native`) features a plugin architecture, built-in type definition and broader platform support. New React Native SDK isn't backwards compatible with maintenance React Native SDK `@amplitude/react-native`. 
+
+To migrate to `@amplitude/analytics-react-native`, update your dependencies and instrumentation.
+
+## Terminology
+
+* `@amplitude/react-native`: Maintenance React Native SDK
+* `@amplitude/analytics-react-native`: New React Native SDK
+
+## Dependency
+
+Uninstall `@amplitude/react-native` by either `yarn remove @amplitude/react-native` or deleting your dependency list in package.json
+
+=== "@amplitude/react-native"
+
+    ```json
+    {
+      "dependencies": {
+        "@amplitude/react-native": "^2"
+      }
+    }
+    ```
+
+Follow [the installation section](../#installation) to add new React Native SDK dependencies.
+
+## Instrumentation
+
+New React Native SDK offers an API to instrument events. To migrate to it, you need to update a few calls. The following sections detail which calls have changed.
+
+### Initialization
+
+`getInstance()` in the maintenance SDK only accepts API key and has been removed. To initialize the SDK, call `init()`, with the user ID and configuration parameters.
+
+=== "@amplitude/react-native"
+
+    ```typescript
+    import { Amplitude } from '@amplitude/react-native';
+
+    Amplitude.getInstance().init(API_KEY)
+    ```
+
+=== "@amplitude/analytics-react-native"
+
+    ```typescript
+    import { init } from '@amplitude/analytics-react-native';
+
+    init(API_KEY, OPTIONAL_USER_ID, config);
+    ```
+
+### Configuration
+
+Maintenance React Native SDK runs on top of the maintenance Android SDK and maintenance iOS SDK and uses settable config options. However new React Native SDK instance has an configuration object in it so that we can set configurations directly on initialization via the instance.
+
+|@amplitude/react-native|@amplitude/analytics-react-native|
+|-|-|
+| `enableCoppaControl()` | NOT SUPPORTED |
+| `disableCoppaControl()` | NOT SUPPORTED |
+| `setAdvertisingIdForDeviceId()` | NOT SUPPORTED |
```

Comment:
For `setAdvertisingIdForDeviceId` and `setAppSetIdForDeviceId`, I think we are still able to support them. but there are no method calls like that. I'm not super familiar with how rn works, but I found we are able to find appSetId and AdvertisingId at [there](https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-react-native/android/src/main/java/com/amplitude/reactnative/AndroidContextProvider.kt#L53-L65). So I feel if you enable to fetch those ids, you can use the plugin to use those ids as deviceId. Please double check in the rn code.

More reference: https://amplitude.atlassian.net/wiki/spaces/GOV/pages/1862926669/2022-08-16+ReactNative-TS+SDK+Sync+on+the+native+code+imports+and+recent+RN-TS+issues

Reply:
I updated `setAdvertisingIdForDeviceId()` to be 

"No configuration to set ADID as device ID. But ADID is still tracked by default as `config.trackingOptions.adid` defaults to `true`. To learn more about how device ID is initialized [here](./#device-id). "

However, I don't think we track app set id as
- app set id is not in HTTP V2 API 
- I searched the codebase and found no `appSetId` elsewhere
- for both latest and legacy Android SDK, we only support `useAppSetIdForDeviceId` but not track app set id as a user property